ngi_reports/utils/entities.py

```diff
@@ -347,6 +363,50 @@ def populate(self, log, organism_names, **kwargs):
                                         'ApplicationVersion': fc_runp.get('ApplicationVersion')
                                         }
 
+            ## collect info of samples and their library prep / LIMS indexes on the FC (only if -b option is set)
+            if kwargs.get('barcode_from_fc'):
+                log.info('\'barcodes_from_fc\' option was given so index sequences for the report will be taken from the flowcell instead of LIMS')
+                preps_samples_on_fc = []
+                list_additional_samples = []
+
+                ## get all samples from flow cell that belong to the project
+                fc_samples = []
+                for fc_sample in fc_details.get('samplesheet_csv'):
+                    if fc_sample.get('Sample_Name').split('_')[0] == self.ngi_id:
+                        fc_samples.append(fc_sample.get('Sample_Name'))
+                
+                ## itterate through all samples in project to identify their prep_ID (only if they are on the flowcell)
+                for sample_ID in list(self.samples):
+                     for prep_ID in list(self.samples.get(sample_ID).preps):
+                        if fcObj.name in self.samples.get(sample_ID).preps.get(prep_ID).seq_fc:
+                            preps_samples_on_fc.append([sample_ID, prep_ID])
+                        else:
+                            continue
+                
+                ## get (if any) samples that are on the fc, but are not recorded in LIMS (i.e. added bc from undet reads)
+                if list(self.samples) == fc_samples:
+                    additional_sample_on_fc = False
```

Comment:
Where else is the variable `additional_sample_on_fc` used?

Reply:
Fair. I intended to use it, but then changed my mind.